src/common/feedGenerator.ts

```diff
@@ -73,24 +78,78 @@ export const whereKeyword = (
   return `EXISTS${query}`;
 };
 
-export const getExcludedAdvancedSettings = async (
+const rawFilterSelect = <T extends ObjectLiteral>(
   con: DataSource,
-  feedId?: string,
-): Promise<Partial<AdvancedSettings>[]> => {
-  if (!feedId) {
-    return [];
-  }
+  name: string,
+  func: (qb: QueryBuilder<T>) => QueryBuilder<T>,
+): QueryBuilder<T> =>
+  con.createQueryBuilder().select('jsonb_agg(res)', name).from(func, 'res');
+
+type RawFiltersData = {
+  settings:
+    | Pick<
+        AdvancedSettings,
+        'id' | 'defaultEnabledState' | 'group' | 'options'
+      >[]
+    | null;
+  feedAdvancedSettings:
+    | Pick<FeedAdvancedSettings, 'advancedSettingsId' | 'enabled'>[]
+    | null;
+  tags: Pick<FeedTag, 'tag' | 'blocked'>[] | null;
+  excludeSources: Pick<FeedSource, 'sourceId'>[] | null;
+  memberships: { sourceId: SourceMember['sourceId']; hide: boolean }[] | null;
+};
 
-  const [settings, feedAdvancedSettings] = await Promise.all([
-    con.getRepository(AdvancedSettings).find(),
-    con.getRepository(FeedAdvancedSettings).findBy({ feedId }),
-  ]);
+const getRawFiltersData = async (
+  con: DataSource,
+  feedId: string,
+  userId: string,
+): Promise<RawFiltersData | undefined> => {
+  const selects = [
+    rawFilterSelect(con, 'settings', (qb) =>
+      qb
+        .select(['id', '"defaultEnabledState"', '"group"', 'options'])
+        .from(AdvancedSettings, 't'),
+    ),
+    rawFilterSelect(con, 'feedAdvancedSettings', (qb) =>
+      qb
+        .select(['"advancedSettingsId"', 'enabled'])
+        .from(FeedAdvancedSettings, 't'),
+    ),
+    rawFilterSelect(con, 'tags', (qb) =>
+      qb.select(['tag', 'blocked']).from(FeedTag, 't').where('"feedId" = $1'),
+    ),
+    rawFilterSelect(con, 'excludeSources', (qb) =>
+      qb
+        .select('"sourceId"')
+        .from(FeedSource, 't')
+        .where('"feedId" = $1')
+        .andWhere('blocked = TRUE'),
+    ),
+    rawFilterSelect(con, 'memberships', (qb) =>
+      qb
+        .select('"sourceId"')
+        .addSelect("COALESCE((flags->'hideFeedPosts')::boolean, FALSE)", 'hide')
+        .from(SourceMember, 't')
+        .where('"userId" = $2'),
+    ),
+  ];
+  const query =
+    'select ' + selects.map((select) => `(${select.getQuery()})`).join(', ');
```

Comment:
Hmm I don't like this part honestly there is however `queryDataSource` helper that will minimize pg-pool connect calls like we do for boot, but it will still use multiple queries. 

Would be interested to load test and see if these 3 options is the best:
- current implementation
- `queryDataSource`
- your implementation with single query

Reply:
yap. i want to generate a report only for this part based on the traces. since it's a child trace i still haven't found a way. i'm not sure it will improve myself either.

reason i use raw queries is that typeorm doesn't allow to build queries without a from statement, but i don't need it here

Reply:
Talking because I am not sure which is the bottleneck here, the multiple queries or just our data being saved in inefficient format across multiple tables.

Reply:
issue is i need to deploy it to test but first i'll try to generate a report

Reply:
As for report, I think its more of a limitation on google trace explorer side no, some other trace interfaces are more powerful.

Reply:
The report always analyzes the parent :triggered:
https://console.cloud.google.com/traces/tasks/2994973240782792936?project=devkit-prod